conda/deprecations.py

```diff
@@ -6,7 +6,33 @@
 from types import ModuleType
 import warnings
 
-from packaging.version import parse, Version
+try:
```

Comment:
I don't think we need this, it rubs me the wrong way that we implement a custom version parser just for `deprecations`.

The `packaging` dependency exists and if not, then we need to find the reason why it's not working and not paper over it.

Reply:
I do like this, I think the packaging import and full Version parsing is overkill for deprecations, and I'm concerned that we have too many dependencies. However I did remove it in the other PR, and if tests pass we can cherry pick that commit here.

Reply:
`packaging` is exclusively used here today but don't we expect to use it more as `distutils` is deprecated?

Reply:
I've split it out into a different PR. Maybe we do, or maybe deprecation is a special case since it has to be accessed so early in conda's initialization cycle.